File: BehavioralDesignPattern/MediatorPattern/python/Driver.py
from ConcreteColleague.User import User
from ConcreteMediator.Chatgroup import Chatgroup


if __name__ == "__main__":
    group = Chatgroup("Test Group")

    user1 = User("Ankit", group)
    user2 = User("Taylor", group)
    user3 = User("Tanmoy", group)

    group.add(user1)
    group.add(user2)
    group.add(user3)

    user1.sendMessage("Hello from Ankit")

File: BehavioralDesignPattern/MediatorPattern/java/Driver.java
package BehavioralDesignPattern.MediatorPattern.java;

import BehavioralDesignPattern.MediatorPattern.java.ConcreteColleague.User;
import BehavioralDesignPattern.MediatorPattern.java.ConcreteMediator.ChatGroup;

public class Driver {
    public static void main(String[] args){
        ChatGroup group = new ChatGroup("Family");

        User user1 = new User("Ankit", group);
        User user2 = new User("Taylor", group);
        User user3 = new User("Tanmoy", group);

        group.add(user1);
        group.add(user2);
        group.add(user3);

        user1.send("Hello this is Ankit");
    }
}
